etl/parse_email_save_attachment/parse_messages.py

```
@@ -103,19 +105,19 @@ def parse_email_from_s3(email_content):
 
     # List of headers to check
     headers_to_check = {
-        'X-SES-Spam-Verdict': 'PASS',
-        'X-SES-Virus-Verdict': 'PASS',
-        'Received-SPF': 'pass',
-        'X-OriginatorOrg': 'austintexas.gov'
+        "X-SES-Spam-Verdict": "PASS",
+        "X-SES-Virus-Verdict": "PASS",
+        "Received-SPF": "pass",
```

The system that Rene and the Maximo team have in place no longer is providing the `X-OriginatorOrg` header. This is unfortunate, but entirely upstream of us, so I'm going to mod the program to reflect that we no longer expect that header.

---

i went back and looked at https://github.com/cityofaustin/dts-maximo-geo-integration/pull/1, and this should be a bug that these headers not meeting our checks didn't cause an Airflow error, right? It has been a while, but I think either way that we should revisit [John's comment here](https://github.com/cityofaustin/dts-maximo-geo-integration/pull/1#pullrequestreview-1906787981) since this seems to be solely relying on having the correct email to get a file into the S3 bucket.

I might also be missing rules set in the SES setup, but I made https://github.com/cityofaustin/atd-data-tech/issues/20493 to collect ideas.

---

If I am understanding correctly, the bug you are describing is that since we have not been getting the X-OriginatorOrg header, we should be seeing an error from the etl and a failure in airflow right?

I checked the most recent log in airflow and see
```
[2025-01-07, 15:00:26 UTC] {docker.py:438} INFO - Hash 5166db58 has not been seen before.
[2025-01-07, 15:00:26 UTC] {docker.py:438} INFO - Email validation condition not met: X-OriginatorOrg is None. Expected: austintexas.gov
[2025-01-07, 15:00:27 UTC] {taskinstance.py:340} ▼ Post task execution logs
[2025-01-07, 15:00:27 UTC] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=maximo_geo_emergency_mgmt_email_parser_production, task_id=parse_email, run_id=scheduled__2025-01-07T14:30:00+00:00, execution_date=20250107T143000, start_date=20250107T150019, end_date=20250107T150027
[2025-01-07, 15:00:27 UTC] {local_task_job_runner.py:266} INFO - Task exited with return code 0
```

reading the code I dont see where an error would be thrown if the header is missing, it just prints the message and then quits.

---

oh but then its a silent failure, isnt it, i was reading it as quitting the loop, like `break` but this is quitting the entire program, not uploading the attachment at all but still coming across as a successful execution!

---

The program was not throwing an error when that header was not present. I think that would be a good addition in the spin-off issue that Mike created. Thank you for that place to collect ideas @mddilley. 

I think that a prudent next step would be to reach out to Rene or David to learn more about what changes they have executed on the upstream portion of this process that sends the email. When we last worked on this, we were presented examples of emails that we needed to handle, and we wrote the program to handle them. It was unbeknownst to us that they have done something that caused that header to not be there. I don't really think that this program could have been written to handle any upstream change that they may have made. I am unsure how it's gone unnoticed that emails were not being parsed for some amount of time since this new system was put in place. 

Also, I think it's important for us to remember that this email receipt system is only intended to be a bridge until a better, non-email based transport for this data is established. Again, we'll have to ask David what the status is of that.

---

Thanks for this discussion. A follow-up question that is gnawing at me a bit is what exactly happened that caused Maximo to end support for XLSX or CSV. or did they never support it? i know that this is "upstream problems" but if we don't ask, who will?